src/cascadia/TerminalSettingsModel/ApplicationState.h

```diff
@@ -44,45 +53,51 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         friend ::Microsoft::Terminal::Settings::Model::JsonUtils::ConversionTrait<Model::WindowLayout>;
     };
 
-    struct ApplicationState : ApplicationStateT<ApplicationState>
+    struct ApplicationState : public ApplicationStateT<ApplicationState>
     {
         static Microsoft::Terminal::Settings::Model::ApplicationState SharedInstance();
 
-        ApplicationState(std::filesystem::path path) noexcept;
+        ApplicationState(const std::filesystem::path& stateRoot) noexcept;
         ~ApplicationState();
 
         // Methods
         void Reload() const noexcept;
         void Reset() noexcept;
 
+        void FromJson(const Json::Value& root, FileSource parseSource) const noexcept;
+        Json::Value ToJson(FileSource parseSource) const noexcept;
+
         // General getters/setters
-        winrt::hstring FilePath() const noexcept;
+        bool IsStatePath(const winrt::hstring& filename);

```

Comment:
Is it really necessary to do `winrt::hstring&`? I thought it internally had some sort of reference copy magic. @lhecker do you know?

Reply:
It's not necessary, but if we can save an AddRef/Release, why not?

Reply:
With the reference in place, MSVC can properly inline this call into the generated WinRT classes.